### PR TITLE
Added syntax for FreeFEM++ and MARTe2 framework

### PR DESCRIPTION
**Pull request by Mandarancio**
Hi,

I add syntax support for the following languages:
 - FreeFEM++ (basically a C++ DSL for FEM simulation)
 - MARTe2 configuration language

thank you very much for your work!

### CODE REVIEW

File: plugins/language_edp.lua [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+    ["seekg"] = "function",
+    ["tellg"] = "function",
+    ["AddLayers"] = "function",
+    ["AffineCG"] = "function",
+    ["AffineGMRES"] = "function",
+    ["BFGS"] = "function",
+    ["EigenValue"] = "function",
+    ["LinearCG"] = "function",
+    ["LinearGMRES"] = "function",
+    ["NLCG"] = "function",
```
Comment (Guldoman) [inline plugins/language_edp.lua:476]: Whitespace at the end.

File: plugins/language_edp.lua [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+    ["Recv"] = "function",
+    ["mpiAlltoall"] = "function",
+    ["mpiGather"] = "function",
+    ["mpiScatter"] = "function",
+    ["mpiReduce"] = "function",
+    ["mpiAllReduce"] = "function",
+    ["mpiReduceScatter"] = "function",
+
+  },
+}
```
Comment (Guldoman) [inline plugins/language_edp.lua:633]: Could you add a newline at the end of the file?

File: plugins/language_marte.lua [hunk 3]
```diff
@@ -0,0 +1,42 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "MARTe",
+  files = { "%.cfg$", "%.marte$" },
```
Comment (Guldoman) [inline plugins/language_marte.lua:6]: I'm a bit reluctant about letting this apply to `.cfg`...

File: plugins/language_marte.lua [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  comment = "//",
+  block_comment = { "/*", "*/" },
+  patterns = {
+    { pattern = { "//", "\n"},                 type = "comment"  },
```
Comment (Guldoman) [inline plugins/language_marte.lua:10]:
```suggestion
    { pattern = "//.*",                        type = "comment"  },
```

**Review by Guldoman — COMMENTED** [review]
The `README` is generated from `manifest.json`, would you mind updating that?

File: plugins/language_marte.lua (hunk 3 above)
Comment (Mandarancio) [reply plugins/language_marte.lua:6]: I agree, I will only use marte extension.

File: plugins/language_edp.lua (hunk 1 above)
Comment (Mandarancio) [reply plugins/language_edp.lua:476]: I think you were referring to the empty line at the end of the function list? if yes now it solved.

**Comment by Mandarancio** [comment]
I also added CUE language support

File: plugins/language_cue.lua [hunk 5]
```diff
@@ -0,0 +1,42 @@
+-- mod-version:3 
+-- lite-xl 2.0
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Cue",
```
Comment (Guldoman) [inline plugins/language_cue.lua:6]: I wonder if this should be uppercase.

Comment (Mandarancio) [reply plugins/language_cue.lua]: You are right, `CUE` should be uppercase, and now I reformatted  `language_edp.cue` using the lsp and should be fine.

File: plugins/language_cue.lua [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  	{ pattern = "//.-\n",               type = "comment"  },
+    { pattern = { '"', '"', '\\' },     type = "string"   },
+    { pattern = { "`", "`", '\\' },     type = "string"   },
+    { pattern = { "'", "'", '\\' },     type = "string"   },
+    { pattern = "0[oO_][0-7]+i?",       type = "number"   },
+    { pattern = "-?0x[%x_]+i?",         type = "number"   },
+    { pattern = "-?%d+_%di?",           type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?i?",   type = "number"   },
+    { pattern = "-?%.?%d+f?i?",         type = "number"   },
+    -- { pattern = "[%a_][%w_]*%.",        type = "function" },
```
[the remark this answers is not in the document]
Comment (Mandarancio) [reply plugins/language_cue.lua]: fixed

File: plugins/language_cue.lua [hunk 7]
```diff
@@ -0,0 +1,42 @@
+-- mod-version:3 
+-- lite-xl 2.0
```
[the remark this answers is not in the document]
Comment (Mandarancio) [reply plugins/language_cue.lua]: fixed

File: plugins/language_cue.lua [hunk 8]
```diff
@@ -0,0 +1,41 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "CUE",
+  files = "%.cue$",
+  comment = "//",
+  patterns = {
+    { pattern = "//*",                       type = "comment" },
```
Comment (Guldoman) [inline plugins/language_cue.lua:9]:
```suggestion
    { pattern = "//.*",                      type = "comment" },
```

**Comment by Guldoman** [comment]
Looks good, thank you!

**Pull request merged**